cashier-sample-google-play-billing/src/main/java/com/getkeepsafe/cashier/sample/googleplaybilling/MainActivity.java

```diff
@@ -0,0 +1,326 @@
+
+        itemsAdapter.setItemListener(new ItemsAdapter.ItemListener() {
+            @Override
+            public void onItemBuy(Item item) {
+                cashier.purchase(MainActivity.this, item.product, purchaseListener);
+            }
+
+            @Override
+            public void onItemUse(final Item item) {
+                progressDialog = new ProgressDialog(MainActivity.this);
+                progressDialog.setIndeterminate(true);
+                progressDialog.setTitle("Consuming item, please wait...");
+                progressDialog.show();
+                new Thread(new Runnable() {
+                    @Override
+                    public void run() {
+                        cashier.consume(item.purchase, consumeListener);
+                    }
+                }).start();
+            }
+
+            @Override
+            public void onItemGetDetails(Item item) {
+                cashier.getProductDetails(item.product.sku(), item.isSubscription, productDetailsListener);
+            }
+        });
+
+        testProduct = Product.create(
+                GooglePlayBillingConstants.VENDOR_PACKAGE,
+                "android.test.purchased",
+                "$0.99",
+                "USD",
+                "Test product",
+                "This is a test product",
+                false,
+                990_000L);
+
+        testProduct2 = Product.create(
+                GooglePlayBillingConstants.VENDOR_PACKAGE,
+                "com.abc.def.123",
+                "$123.99",
+                "USD",
+                "Test product 2",
+                "This is another test product",
+                false,
+                123990_000L);
+
+        // For testing certain products
+        FakeGooglePlayBillingApi.addTestProduct(testProduct);
+        FakeGooglePlayBillingApi.addTestProduct(testProduct2);
+
+        initCashier();
+
+        queryPurchases.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                refreshItems();
+            }
+        });
+
+        useFake.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                initCashier();
+            }
+        });
+
+    }
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        cashier.dispose();
+    }
+
+    private void initCashier() {
+        if (cashier != null) {
+            cashier.dispose();
+        }
+
+        new Thread() {
+            public void run() {
+                Vendor vendor;
+                if (useFake.isChecked()) {
+                    vendor = new GooglePlayBillingVendor(
+                            new FakeGooglePlayBillingApi(MainActivity.this,
+                                    FakeGooglePlayBillingApi.TEST_PUBLIC_KEY));
+                } else {
+                    vendor = new GooglePlayBillingVendor();
```

Comment:
Can you explain how this works without the private key? I did not know that you can query Google Play without a private key.

Reply:
Please disregard the comment. Google Play Billing has the copy of the key in its servers. The key does not need to be passed to the Billing client.